app/models/bookmark.rb

```diff
@@ -103,6 +103,20 @@ def self.count_per_day(opts = nil)
       .order('date(bookmarks.created_at)')
       .count
   end
+
+  ##
+  # Deletes bookmarks that are attached to posts/topics that were deleted
+  # more than X days ago. We don't delete bookmarks instantly when a post/topic
+  # is deleted so that there is a grace period to un-delete.
+  def self.cleanup!
+    grace_time = 3.days.ago
```

Comment:
This could be in a site setting but I didn't really see the value in making this configurable.